c++11/include/opentracing/util.h

```diff
@@ -0,0 +1,58 @@
+#ifndef OPENTRACING_UTIL_H
+#define OPENTRACING_UTIL_H
+
+#include <opentracing/version.h>
+#include <chrono>
+#include <opentracing/martinmoene_expected/expected.hpp>
```

Comment:
[Expected](http://open-std.org/JTC1/SC22/WG21/docs/papers/2017/p0323r2.pdf) is meant to support non-exception error handling. It's used to deal with invalid_contexts, corrupted spans, invalid carrier, etc. C++11 added [std::error_code](http://en.cppreference.com/w/cpp/error/error_code) that provides a standard way to make descriptive user errors; expected gives you a way to return such errors from a function.

The expected implementation used is just a small header-only library that's bundled along with the code (like mapbox::variant). There aren't any external dependencies.